src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredMen = people
+    .filter((person) => {
+      return century !== undefined
+        ? Math.ceil(person.died / 100) === century
+          && person.sex === 'm'
+        : person.sex === 'm';
```

Comment:
Instead of having duplicated conditions, create two booleans:
```js
const isMan = person.sex === 'm';
const isFromCentury = century ? <calculation> === century : true;
```
Return result of AND operator.
Use this principle everywhere. It's clearer and easy to understand.